src/annotator/sidebar.js

```diff
@@ -0,0 +1,326 @@
+import $ from 'jquery';
+import Hammer from 'hammerjs';
+
+import annotationCounts from './annotation-counts';
+import sidebarTrigger from './sidebar-trigger';
+import events from '../shared/bridge-events';
+import features from './features';
+
+import { ToolbarController } from './toolbar';
+
+// Minimum width to which the frame can be resized.
+const MIN_RESIZE = 280;
+
+// TODO - Convert this to an ES import once `host.coffee` is converted.
+// @ts-expect-error
+const Host = require('./host');
+
+const defaultConfig = {
+  Document: {},
+  TextSelection: {},
+  BucketBar: {
+    container: '.annotator-frame',
+  },
+};
+
+export default class Sidebar extends Host {
+  constructor(element, config) {
+    if (config.theme === 'clean' || config.externalContainerSelector) {
+      delete config.pluginClasses.BucketBar;
+    }
+
+    super(element, { ...defaultConfig, ...config });
+
+    if (
+      config.openSidebar ||
+      config.annotations ||
+      config.query ||
+      config.group
+    ) {
+      this.on('panelReady', () => this.show());
+    }
+
+    if (this.plugins.BucketBar) {
+      this.plugins.BucketBar.element.on('click', () => this.show());
+    }
+
+    // Set up the toolbar on the left edge of the sidebar.
+    const toolbarContainer = document.createElement('div');
+    this.toolbar = new ToolbarController(toolbarContainer, {
+      createAnnotation: () => this.createAnnotation(),
+      setSidebarOpen: open => (open ? this.show() : this.hide()),
+      setHighlightsVisible: show => this.setAllVisibleHighlights(show),
+    });
+    this.toolbar.useMinimalControls = config.theme === 'clean';
+
+    if (this.frame) {
+      // If using our own container frame for the sidebar, add the toolbar to it.
+      this.frame.prepend(toolbarContainer);
+      this.toolbarWidth = this.toolbar.getWidth();
+    } else {
+      // If using a host-page provided container for the sidebar, the toolbar is
+      // not shown.
+      this.toolbarWidth = 0;
+    }
+
+    this._gestureState = {
```

Comment:
In scanning this, it's not clear why the values for the properties would be numbers. I'll keep reading the module and see if it becomes clearer...

Reply:
Upon further reading, I believe these are pixel values. Perhaps a comment here?